### ISSUE TITLE
Ladybird dropdown menu display issue

### ISSUE THREAD

**Issue by jordanmartell0** [SerenityOS/serenity#23920]
Hello, I'm a University of Utah student working on Ladybird as part of my final project for CS 4560 Web Browser Internals. While using the browser, I found a bug in how it displays the arrow of a dropdown menu in certain web pages (pictured below are how Chrome displays the arrow and how Ladybird displays the same arrow)

Chrome:
 
![image](https://github.com/SerenityOS/serenity/assets/78096738/ba915c82-c52d-4cb2-9a46-e7be3d219ca1)

Ladybird:

![image](https://github.com/SerenityOS/serenity/assets/78096738/792da476-9d2b-44d8-9512-93958b6c440e)

I minimized the bug down to this simple html and css, which Chrome renders as a diagonal line and Ladybird renders as a solid box:

```
<style>
  .dropdown-arrow-icon{
      transform:rotate(-45deg);
      background:#202124;
      height:7px;
  }
</style>

<div class="dropdown-arrow-icon"></div>
```

Chrome:
 
![image](https://github.com/SerenityOS/serenity/assets/78096738/41b57173-25a1-4f32-85d8-df51832641c6)

Ladybird:
 
![image](https://github.com/SerenityOS/serenity/assets/78096738/e4893431-002a-45ad-9616-06e52a0af65e)

I am not really sure where to start as far as resolving this issue, but I am interested in taking a look and would appreciate any pointers on how to proceed. Thanks!

**Comment by kalenikaliaksandr** [comment]
we do not have support for rotation transform in CPU painter (used by default) yet https://github.com/SerenityOS/serenity/blob/master/Userland/Libraries/LibWeb/Painting/CommandExecutorCPU.cpp#L156

you could start from figuring if we can rotate a bitmap using LibGfx painter https://github.com/SerenityOS/serenity/blob/master/Userland/Libraries/LibGfx/Painter.cpp

**Comment by nico** [comment]
(@MacDue has some local branch for that I think, ref https://discord.com/channels/830522505605283862/830526926569209917/1224462326250344531)

**Comment by MacDue** [comment]
Yes, I have a prototype that I think would fix this which would rotate the actual draw call (and paint this via the path rasterizer). I've not got too far with the implementation yet, I've been a little busy, but if I get a branch going I'd be happy for some help in this area.

I would really like to avoid trying to do this by rotating bitmaps, because that's always going to introduce artifacts (and is fairly slow too, at least in LibGfx).

**Comment by OfficialPixelBrush** [comment]
Wasn't there a fast way to fake rotation? Something about skewing?

**Comment by MacDue** [comment]
Lot of work left to do, but this is how the reproducer looks on with the new painter:
![image](https://github.com/SerenityOS/serenity/assets/11597044/04e504d1-5626-455e-9362-7c6280c5bd1f)

The branch is: https://github.com/MacDue/serenity/tree/new_painting_hacks, very WIP, and I've not had time to work on it much recently :pensive:
